versions/draft-3/SPEC.md

```diff
@@ -724,26 +724,22 @@ $task_sections = ($command | $runtime | $task_output | $parameter_meta | $meta)+
 
 ### Command Section
 
-```
-$command = 'command' $ws* '{' (0xA | 0xD)* $command_part+ $ws+ '}'
-$command = 'command' $ws* '<<<' (0xA | 0xD)* $command_part+ $ws+ '>>>'
-```
+A command is a *task section* that starts with the keyword 'command'. It defines a shell command which will be run in the appropriate environment after all of the inputs are staged and before the outputs are evaluated. 
 
-A command is a *task section* that starts with the keyword 'command', and is enclosed in curly braces or `<<<` `>>>`.  The body of the command specifies the literal command line to run with placeholders (`$command_part_var`) for the parts of the command line that needs to be filled in.
+There are two styles of command body, enclosed in curly braces `{` `}` or triple-angle brackets `<<<` `>>>`.  These command bodies specify a command line script to run and may contain placeholders for WDL expressions in order to customize the script. The usual WDL parsing rules are changed inside the command section: with the exception of expression placeholders, everything (even things that look like WDL comments) are interpreted as a literal string to be included in the command script.
 
-#### Command Parts
+Expression placeholders are denoted by `${...}` or `~{...}` depending on whether they appear in a `command { }` or `command <<< >>>` body styles:
 
-```
-$command_part = $command_part_string | $command_part_var
-$command_part_string = ^'${'+
-$command_part_var = '${' $var_option* $expression '}'
-```
+|Command Body Style|Placeholder Style|
+|---|---|
+|`command { ... }`|`${}` or `~{}`|
```

Comment:
Why not just have `~{}`?

Reply:
My thought was that the `command {...}` syntax is already not good enough for bash syntax so we might as well allow users the `${...}` style placeholders. 

Eg even if `${...}` is not a WDL placeholder we still can't use it in a `command {...}` section because the final `}` is going to incorrectly close the command section.

Reply:
Leaving aside whether "you must have matching numbers of `{`s and `}`s in your command block" is a requirement we'd want to add to WDL...

The more fundamental problem with that idea is that this "counting" would have to happen in the lexer, and "keeping track of opening and closing braces" is something that most lexers can't do (you can't do it with a FSA for the same reason you can't do it with a regex).

As it turns out Hermes has an "infinite" set of states with its stacks so I guess it might be able to handle that, but it would mess up the intellij, sublime, github, etc integration which are all FSA based.